frontend/src/@types/auth.d.ts

```diff
@@ -0,0 +1 @@
+export type SocialProvider = 'amazon' | 'apple' | 'facebook' | 'google';
```

Comment:
独自の宣言に置換した意図としては、どのようなものがありますか？

Reply:
xstateをinstallした際に、peer dependencyでamplifyがV4参照していたらしく型エラーが発生したため個別で定義しましたmm

Reply:
ありがとうございます！コメント記載お願いします！

Reply:
```
Due to peer dependency issues arising from the introduction of xstate V5, I implemented it by directly defining types instead of through the library.
```

上記追加しておきます！